### PR TITLE
Merge reacher_env from rjzp-temp branch

### LINKED ISSUE #8: Merge reacher from rjzp-temp branch to master
rjzp-temp has a reacher that works better for experiment.

### PR DESCRIPTION
**Pull request by utkarshjp7**
This PR fixes #8.

### CODE REVIEW

**Comment by utkarshjp7** [comment]
@zhanpenghe Moving our slack conversation here.

The following test fails if I merge the base class `SawyerEnv`. The test passes otherwise. 

`def test_reacher(self):
        """Testing for reacher."""
        tasks = [(0.3, -0.3, 0.30), (0.3, 0.3, 0.30)]

        env = ReacherEnv(goal_position=tasks[0])
        step_env(env, n=5, render=True)`

**Comment by utkarshjp7** [comment]
@ryanjulian Heija is working on integrating the pusher environment.

**Comment by utkarshjp7** [comment]
@zhanpenghe The base class SawyerEnv is now merged.

File: sawyer/mujoco/sawyer_env.py [hunk 1]
```diff
@@ -149,45 +372,80 @@ def observation_space(self):
...
+            next_pos = np.clip(
+                a + curr_pos[2:],
+                self.joint_position_space.low[2:],
+                self.joint_position_space.high[2:]
+            )
+            for _ in range(4):
+                self.sim.data.ctrl[2:] = next_pos
+                self.sim.forward()
+                self.sim.step()
+            # d = np.linalg.norm(self.joint_positions[2:] - next_pos, axis=-1)
```
Comment (zhanpenghe) [inline sawyer/mujoco/sawyer_env.py]: Please remove commented-out codes.

File: sawyer/mujoco/sawyer_env.py [hunk 2]
```diff
@@ -149,45 +372,80 @@ def observation_space(self):
...
+        # collision detection
+        if in_collision:
+            r -= self._collision_penalty
+            if self._terminate_on_collision:
+                done = True
+
         if self._is_success:
             r = self._completion_bonus
-            done = True
+            # done = True
```
Comment (zhanpenghe) [inline sawyer/mujoco/sawyer_env.py]: Commented-out code.

File: sawyer/mujoco/sawyer_env.py [hunk 3]
```diff
@@ -198,10 +456,10 @@ def set_gripper_state(self, state):
         # 1 = open, -1 = closed
         self.gripper_state = state
         state = (state + 1.) / 2.
-        self.sim.data.ctrl[:] = np.array([state * 0.020833, -state * 0.020833])
-        for _ in range(3):
-            self.sim.step()
-        self.sim.forward()
+        self.sim.data.ctrl[:2] = np.array([state * 0.020833, -state * 0.020833])
+        # for _ in range(3):
```
Comment (zhanpenghe) [inline sawyer/mujoco/sawyer_env.py]: Commented-out code.

File: sawyer/mujoco/sawyer_env.py [hunk 4]
```diff
@@ -235,60 +493,91 @@ def get_obs(self):
         achieved_goal = self._achieved_goal_fn(self)
         desired_goal = self._desired_goal_fn(self)
 
-        achieved_goal_qpos = np.concatenate((achieved_goal, [1, 0, 0, 0]))
-        self.sim.data.set_joint_qpos('achieved_goal:joint', achieved_goal_qpos)
-        desired_goal_qpos = np.concatenate((desired_goal, [1, 0, 0, 0]))
-        self.sim.data.set_joint_qpos('desired_goal:joint', desired_goal_qpos)
+        # achieved_goal_qpos = np.concatenate((achieved_goal, [1, 0, 0, 0]))
```
Comment (zhanpenghe) [inline sawyer/mujoco/sawyer_env.py]: Commented-out code.

File: sawyer/mujoco/sawyer_env.py [hunk 5]
```diff
@@ -73,19 +212,87 @@ def __init__(self,
...
 
-    def _env_setup(self):
+    def env_setup(self):
         reset_mocap_welds(self.sim)
         self.sim.forward()
 
+    def render(self, mode="human"):
+        viewer = self.get_viewer()
+        viewer.add_marker(pos=np.array(self._desired_goal), label="goal", size=0.01 * np.ones(3),)
+        # GOALS = [
```
Comment (zhanpenghe) [inline sawyer/mujoco/sawyer_env.py]: Removed commented out codes.

File: sawyer/mujoco/sawyer_env.py [hunk 6]
```diff
@@ -95,20 +302,31 @@ def set_gripper_position(self, position):
             reset_mocap2body_xpos(self.sim)
             self.sim.data.mocap_quat[:] = np.array([0, 1, 0, 0])
             self.sim.data.set_mocap_pos('mocap', position)
-        self.sim.forward()
+        # self.sim.forward()
```
Comment (zhanpenghe) [inline sawyer/mujoco/sawyer_env.py]: Commented-out code.

**Pull request merged**